### PR TITLE
Inventory stock levels Implemented - Issue #26 

### PR DESCRIPTION
**Pull request by yousefenein**
Completed frontend and backend for Inventory stock levels. NextUI used for the front end, stock level status displayed based on mock thresholds (to be adjusted).  Filtering options implemented as well as ability to sort based on requested columns. 

Left to do : 

1) integrate alerts features - (already completed by James, looking to fully integrate tomorrow) 
2) Add a feature to add items into inventory (TBD later issue #48)

### CODE REVIEW

File: WarehousePilot_app/frontend/src/components/inventory-stock/delete.tsx
Comment (Irisvella) [inline WarehousePilot_app/frontend/src/components/inventory-stock/delete.tsx:1]: I don't see this delete option in the table

File: WarehousePilot_app/frontend/src/components/inventory-stock/edit.tsx
Comment (Irisvella) [inline WarehousePilot_app/frontend/src/components/inventory-stock/edit.tsx:1]: I don't see this edit function

**Review by Irisvella — APPROVED** [review]
- Correctly pulls inventory stock from backend and displays 
- Displays information quickly and the page responds quickly to user action

Things to refactor for Iteration 7:
- Sidebar/Header missing
- "add item" not working
- stock item "actions" not working
- rename search to "search by SKU Color ID" 
- Sorting only works per page rather than globally 
- An improvement could be to display more items per page 
- Add testing

**Comment by sarahabellard** [comment]
Some of the issues like sorting might be able to be fixed using MUI and will also ensure uniformity. More discussion will be needed during meeting on this. Refracturing and comments is set to be addressed early next iteration. Data being displayed works as expected.

**Review by sarahabellard — APPROVED** [review]
image

**Comment by yousefenein** [comment]
> * Correctly pulls inventory stock from backend and displays
> * Displays information quickly and the page responds quickly to user action
> 
> Things to refactor for Iteration 7:
> 
> * Sidebar/Header missing
> * "add item" not working
> * stock item "actions" not working
> * rename search to "search by SKU Color ID"
> * Sorting only works per page rather than globally
> * An improvement could be to display more items per page
> * Add testing - Remaining to do 


These are now completed. 

https://github.com/user-attachments/assets/ddbe7842-9cc2-4dd9-a4ce-eb47d7556ba8

**Pull request merged**